### PR TITLE
feat: update dependencies & add new rules

### PR DESCRIPTION
**Pull request by MichaelDeBoey**
- [`logical-assignment-operators`](https://eslint.org/docs/latest/rules/logical-assignment-operators)
- [`no-empty-static-block`](https://eslint.org/docs/latest/rules/no-empty-static-block)
- [`no-new-native-nonconstructor`](https://eslint.org/docs/latest/rules/no-new-native-nonconstructor)
- [`import/consistent-type-specifier-style`](https://github.com/import-js/eslint-plugin-import/blob/main/docs/rules/consistent-type-specifier-style.md)
- [`import/no-empty-named-blocks`](https://github.com/import-js/eslint-plugin-import/blob/main/docs/rules/no-empty-named-blocks.md)
- [`jsx-a11y/anchor-ambiguous-text`](https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/blob/main/docs/rules/anchor-ambiguous-text.md)
- [`jsx-a11y/no-aria-hidden-on-focusable`](https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/blob/main/docs/rules/no-aria-hidden-on-focusable.md)
- [`jsx-a11y/prefer-tag-over-role`](https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/blob/main/docs/rules/prefer-tag-over-role.md)
- [`react/no-object-type-as-default-prop`](https://github.com/jsx-eslint/eslint-plugin-react/blob/master/docs/rules/no-object-type-as-default-prop.md)
- [`react/sort-default-props`](https://github.com/jsx-eslint/eslint-plugin-react/blob/master/docs/rules/sort-default-props.md)

### CODE REVIEW

File: import.js [hunk 1]
```diff
@@ -17,6 +17,7 @@ module.exports = {
     'import/ignore': ['node_modules', '.json$', '.(scss|less|css|styl)$'],
   },
   rules: {
+    'import/consistent-type-specifier-style': ['error', 'prefer-top-level'],
```
Comment (kentcdodds) [inline import.js:20]:
```suggestion
    'import/consistent-type-specifier-style': ['error', 'prefer-inline'],
```

File: jsx-a11y.js [hunk 2]
```diff
@@ -17,6 +17,7 @@ module.exports = {
   rules: {
     'jsx-a11y/accessible-emoji': 'error',
     'jsx-a11y/alt-text': 'warn',
+    'jsx-a11y/anchor-ambiguous-text': 'off',
```
Comment (kentcdodds) [inline jsx-a11y.js:20]:
```suggestion
    'jsx-a11y/anchor-ambiguous-text': 'warn',
```

File: react.js [hunk 3]
```diff
@@ -102,6 +102,7 @@ module.exports = {
     'react/no-is-mounted': 'error',
     'react/no-multi-comp': 'off',
     'react/no-namespace': 'error',
+    'react/no-object-type-as-default-prop': 'error',
```
Comment (kentcdodds) [inline react.js:105]:
```suggestion
    'react/no-object-type-as-default-prop': 'off',
```

**Review by kentcdodds — CHANGES_REQUESTED** [review]
Thanks! I'm going to make just a few changes here.

File: react.js (hunk 3 above)
Comment (MichaelDeBoey) [reply react.js]: Why are you disabling this rule?
It could cause bugs if we don't follow the guidelines of the rule.

**Pull request merged**